include/algorithms/synthesis/encoding.hpp

```diff
@@ -0,0 +1,159 @@
+#ifndef ENCODING_HPP
+#define ENCODING_HPP
```

Comment:
try to remain consistent with header guards/`#pragma once` when introducing new files.
I'd personally prefer `#pragma once` everywhere over header guards.

Reply:
done 👍🏼